Update meraki for api version 2.1

liptonj opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

liptonj committed:
Updated the meraki component for version 2.1 of scanning api also added the new parameter of name and assied it to the attr dict as well as the device_tracker name value.   Added RSSI to the attr dict

epenet committed:
Remove unused constant from auth (#72953)

MartinHjelmare commented on homeassistant/components/meraki/device_tracker.py:
```diff
@@ -90,6 +90,7 @@ async def post(self, request):
 
     @callback
     def _handle(self, hass, data):
+        ap_mac = data.get("apMac","none")
```
This is never used.

MartinHjelmare commented on homeassistant/components/meraki/device_tracker.py:
```diff
@@ -111,6 +112,7 @@ def _handle(self, hass, data):
                 gps_location = (lat, lng)
 
             attrs = {}
+            attrs["name"] = i.get("name", "None")
```
Don't use `"None"` as default. Use `None`.
```suggestion
            attrs["name"] = i.get("name")
```

MartinHjelmare replied:
We shouldn't set name both in `attrs` and as the name parameter.

MartinHjelmare replied:
How is the name parameter set by the api? Is it always included in the requests

MartinHjelmare commented on homeassistant/components/meraki/device_tracker.py:
```diff
@@ -123,10 +125,15 @@ def _handle(self, hass, data):
+   
+                
+            if i.get("rssi", False):
+                attrs["rssi"] = i["rssi"]
```
We shouldn't add RSSI as a state attribute. This measurement is known for spamming the state machine.

ronytomen commented:
Addresses issue: Meraki API Version #72935

liptonj committed:
Updated meraki device tracker to use ap_mac value set to ap_mac in attrs
change the name value to only set in parameters of the device
removed RSSI per suggestion,but this could be a value to track when trouble shooting iot devices
set the device_name to a default of the mac address if no devices name is sent over via api

liptonj committed:
Updated meraki device tracker to use ap_mac value set to ap_mac in attrs
change the name value to only set in parameters of the device
removed RSSI per suggestion,but this could be a value to track when trouble shooting iot devices
set the device_name to a default of the mac address if no devices name is sent over via api

liptonj committed:
Merge branch 'home-assistant:dev' into dev

liptonj committed:
Updated meraki device tracker to use ap_mac value set to ap_mac in attrs
change the name value to only set in parameters of the device
removed RSSI per suggestion,but this could be a value to track when trouble shooting iot devices
set the device_name to a default of the mac address if no devices name is sent over via api
